sake/redis.py

```diff
@@ -1182,17 +1363,60 @@ async def set_invite(
             expire_time = int(self.metadata.get("expire_invite", DEFAULT_INVITE_EXPIRE))
 
         else:
-            expire_time = _convert_expire_time(expire_time)
+            expire_time = utility.convert_expire_time(expire_time)
 
         # Aioredis treats keys and values as type invariant so we want to ensure this is a str and not a class which
         # subclasses str.
-        await client.set(str(invite.code), data, pexpire=expire_time)
+        await self.get_connection(ResourceIndex.INVITE).set(str(payload["code"]), payload, pexpire=expire_time)
+
+        if target_user := payload.get("target_user"):
+            await self._try_set_user(target_user)
+
+        if inviter := payload.get("inviter"):
+            await self._try_set_user(inviter)
 
-        if invite.target_user is not None:
-            await self._optionally_set_user(invite.target_user)
 
-        if invite.inviter is not None:
-            await self._optionally_set_user(invite.inviter)
+class _OwnIDStore:
+    __slots__: typing.Sequence[str] = ("_event", "_is_waiting", "_rest", "value")
+
+    KEY: typing.Final[typing.ClassVar[str]] = "OWN_ID"
+
+    def __init__(self, rest: hikari_traits.RESTAware, /) -> None:
+        self._event = asyncio.Event()
```

Comment:
add an asyncio.get_running_loop call here to ensure this is only ever used in a running loop